spring/src/main/java/io/kubernetes/client/spring/extended/controller/KubernetesReconcilerProcessor.java

```diff
@@ -54,20 +55,27 @@ public int getOrder() {
   }
 
   @Override
-  public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
-      throws BeansException {
-    String[] names = beanFactory.getBeanNamesForType(Reconciler.class);
-    for (String name : names) {
-      Reconciler reconciler = (Reconciler) beanFactory.getBean(name);
-      KubernetesReconciler kubernetesReconciler =
-          reconciler.getClass().getAnnotation(KubernetesReconciler.class);
-      String reconcilerName = kubernetesReconciler.value();
+  public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) {
+    this.beanDefinitionRegistry = registry;
+  }
+
+  @Override
+  public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) {
+    for (String reconcilerName : beanFactory.getBeanNamesForType(Reconciler.class)) {
+
+      Supplier<KubernetesControllerFactory> kubernetesControllerFactorySupplier =
```

Comment:
I guess this can be inlined. It's a matter of taste though.

Reply:
Agree it's opinionated. I prefer readability vs. compact code.
By the way the PR is finally green... :)